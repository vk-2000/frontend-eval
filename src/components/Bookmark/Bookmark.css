button {
    background: transparent;
    border: none;
    height: 30px;
    width: 30px;
}
button img {
    height: 100%;
    width: 100%;
    filter: invert(81%) sepia(19%) saturate(7366%) hue-rotate(311deg) brightness(109%) contrast(91%);
}
