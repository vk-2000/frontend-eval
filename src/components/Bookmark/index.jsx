import React from 'react';
import PropTypes from 'prop-types';
import solidBookmark from '../../assets/icons/bookmark-solid.svg';
import regularBookmark from '../../assets/icons/bookmark-regular.svg';
import './Bookmark.css';

const Bookmark = ({ isBookmarked, onChange }) => (
  <button data-testid="btn-bookmark" onClick={(e) => { e.stopPropagation(); onChange(); }} type="button">
    <img src={isBookmarked ? solidBookmark : regularBookmark} alt="bookmark" />
  </button>
);

export default Bookmark;

Bookmark.propTypes = {
  isBookmarked: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired,
};
