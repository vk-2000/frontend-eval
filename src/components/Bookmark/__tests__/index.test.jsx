import { fireEvent, render } from '@testing-library/react';
import Bookmark from '..';

describe('Bookmark', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Bookmark isBookmarked={false} onChange={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render solid bookmark when isBookmarked is true', () => {
    const { getByAltText } = render(<Bookmark isBookmarked onChange={() => {}} />);
    const bookmark = getByAltText('bookmark');
    expect(bookmark.src).toContain('bookmark-solid.svg');
  });
  it('should render regular bookmark when isBookmarked is false', () => {
    const { getByAltText } = render(<Bookmark isBookmarked={false} onChange={() => {}} />);
    const bookmark = getByAltText('bookmark');
    expect(bookmark.src).toContain('bookmark-regular.svg');
  });
  it('should call onChange when bookmark is clicked', () => {
    const mockOnChange = jest.fn();
    const { getByTestId } = render(<Bookmark isBookmarked onChange={mockOnChange} />);
    const button = getByTestId('btn-bookmark');
    fireEvent.click(button);
    expect(mockOnChange).toHaveBeenCalled();
  });
});
