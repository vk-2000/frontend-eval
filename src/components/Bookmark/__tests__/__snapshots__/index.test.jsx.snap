// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookmark should render correctly 1`] = `
<DocumentFragment>
  <button
    data-testid="btn-bookmark"
    type="button"
  >
    <img
      alt="bookmark"
      src="bookmark-regular.svg"
    />
  </button>
</DocumentFragment>
`;
